CONSOLE-2479: Add YAML Sample for QuickStart CRD

jhadvig opened the pull request:
Based on the discussion with @bmignano opening a PR thats adding an yaml example for the ConsoleQuickStart CRD.
The example QS should show the user how to deploy application from the `openshift/hello-openshift` image.
It contains two steps:
- deploy the image
- check the deployed app is running

I know we originally discussed that the second step should be to scale up the deployment, but was thinking about it and checking if the app is up by visiting its route should be good enough for a second step.

/assign @spadgett @bmignano

jhadvig commented:
/retest

spadgett commented on frontend/public/models/yaml-templates.ts:
```diff
@@ -1243,6 +1243,60 @@ metadata:
+        2. Go to the **+Add** page in the navigation.
+        3. Using the project drop-down list, select the project you would like to deploy the image in.
+        4. Click **Container Image** to deploy an image.
+        5. Type **openshift/hello-openshift** into the **Image name from external registry** field.
```
We'll want to use `registry.redhat.io/openshift4/ose-hello-openshift-rhel8`. (Note to QE: This image will only be available after 4.7 GA.)

spadgett commented on frontend/public/models/yaml-templates.ts:
```diff
@@ -1243,6 +1243,60 @@ metadata:
+        4. Click **Container Image** to deploy an image.
+        5. Type **openshift/hello-openshift** into the **Image name from external registry** field.
+        6. Scroll down and click **Create**.
+        The **Topology** view will load with your **hello-openshift** application. The application is represented by the light grey area with the white border. The deployment is a white circle.
```
If we use the image above, I suspect the default name will change. We might want to have the user type `hello-openshift` as the name in the steps above.

spadgett reviewed (commented):
I realize this is just a template, but we should probably still use `registry.redhat.io/openshift4/ose-hello-openshift-rhel8` as the image. (See https://bugzilla.redhat.com/show_bug.cgi?id=1766287)

bmignano commented:
Thanks for the quick turnaround, Jakub! This looks good to me, tagging @abigaeljamie for a content review

abigaeljamie commented on frontend/public/models/yaml-templates.ts:
```diff
@@ -1243,6 +1243,60 @@ metadata:
+      review:
+        instructions: |-
+          #### To verify your application is running:
+          1. Make sure you are in the new tab.
```
```suggestion
          In the new tab, do you see **Hello OpenShift!** ?
```

abigaeljamie commented on frontend/public/models/yaml-templates.ts:
```diff
@@ -1243,6 +1243,60 @@ metadata:
+          1. Make sure you are in the new tab.
+          2. Do you see **Hello OpenShift!** text in the new tab?
+      summary:
+        success: Your have run **hello-openshift** application!
```
```suggestion
        success: Great work! You ran the **hello-openshift** application.
```

abigaeljamie replied:
I don't think we'll have congratulatory messages in quick starts anymore, but I don't know for sure.

jhadvig replied:
>  I don't think we'll have congratulatory messages in quick starts anymore, but I don't know for sure.

then how about using:
`success: You have deployed **hello-openshift** image.`

abigaeljamie replied:
Oh, I meant that I don't think any message will appear at all. But if a message does appear, it's fine to just use what we have ("Great work! You ran the hello-openshift application").

abigaeljamie commented on frontend/public/models/yaml-templates.ts:
```diff
@@ -1243,6 +1243,60 @@ metadata:
+      summary:
+        success: Your have run **hello-openshift** application!
+        failed: Try the steps again.
+  conclusion: Your example application is deployed and ready!
```
```suggestion
  conclusion: Your example application is deployed and ready.
```

abigaeljamie reviewed (changes requested):
I'm so glad something like this will exist! I left edits, and I can do another review if you like once the feedback is applied :) Cheers!

jhadvig commented on frontend/public/models/yaml-templates.ts:
```diff
@@ -1243,6 +1243,60 @@ metadata:
+  description: Example application.
+  prerequisites:
+  introduction: '### This Quick Start shows you how to deploy an existing image to OpenShift.'
+  tasks:
```
This is CRD `field` that we have to use.

jhadvig commented:
@abigaeljamie thanks for review :) Addressed most of your comments and added the link for the QS guidline into the into the `introduction` field.
PTAL

jhadvig commented:
/retest

abigaeljamie commented:
> @abigaeljamie thanks for review :) Addressed most of your comments and added the link for the QS guidline into the into the `introduction` field.
> PTAL

Yay! Thank you!

jhadvig commented:
@abigaeljamie @bmignano PTAL

jhadvig commented on frontend/public/models/yaml-templates.ts:
```diff
@@ -1243,6 +1243,60 @@ metadata:
+        4. Click **Container Image** to deploy an image.
+        5. Type **openshift/hello-openshift** into the **Image name from external registry** field.
+        6. Scroll down and click **Create**.
+        The **Topology** view will load with your **hello-openshift** application. The application is represented by the light grey area with the white border. The deployment is a white circle.
```
makes sense to leave it out. will remove

jhadvig commented on frontend/public/models/yaml-templates.ts:
```diff
@@ -1243,6 +1243,60 @@ metadata:
+      review:
+        instructions: |-
+          #### To verify the image was successfully deployed:
+          1. Do you see a **hello-openshift** deployment?
```
Here is the topology view. By default the Deployment's name is `hello-openshift` and the application name is `hello-openshift-app`. Those names are generated automatically. Update the names accordingly. 
<img width="1917" alt="Screenshot 2020-12-16 at 17 27 02" src="https://user-images.githubusercontent.com/1668218/102376429-ea7a7a80-3fc3-11eb-9422-13238e366dd3.png">

jhadvig commented on frontend/public/models/yaml-templates.ts:
```diff
@@ -1243,6 +1243,60 @@ metadata:
+    - title: Running the deployed application
+      description: |-
+        ### To run **hello-openshift** deployment application:
+        1. Click on the deployment's white circle in the **Topology** view.
```
the outer is the application, the inner is the deployment.

abigaeljamie commented on frontend/public/models/yaml-templates.ts:
```diff
@@ -1243,6 +1244,60 @@ metadata:
+      summary:
+        success: You have deployed **hello-openshift** image.
+        failed: Try the steps again.
+  conclusion: Your example **hello-openshift-app** application, using **hello-openshift** image is deployed and ready.
```
```suggestion
  conclusion: Your example **hello-openshift-app** application, using the **hello-openshift** image, is deployed and ready.
```

abigaeljamie commented on frontend/public/models/yaml-templates.ts:
```diff
@@ -1243,6 +1244,60 @@ metadata:
+          In the new tab, do you see **Hello OpenShift!** ?
+        failedTaskHelp: This task isn’t verified yet. Try the task again.
+      summary:
+        success: You have deployed **hello-openshift** image.
```
```suggestion
        success: Great work! You deployed the **hello-openshift** image.
``` 
We won't be using any success messages moving forward. So if possible, this can be removed (but if you can't remove it, I went ahead and edited it to match our other success message).

abigaeljamie commented on frontend/public/models/yaml-templates.ts:
```diff
@@ -1243,6 +1244,60 @@ metadata:
+    - title: Run the deployed application
+      description: |-
+        ### To run the **hello-openshift** deployment application, follow these steps:
+        1. In the **Topology** view, click the deployment's white circle.
```
Is there any other identifier we can use in addition to the color so that we align with accessibility guidelines?

abigaeljamie commented on frontend/public/models/yaml-templates.ts:
```diff
@@ -1243,6 +1244,60 @@ metadata:
+  durationMinutes: 5
+  description: Deploy an example application from an image.
+  introduction: |-
+    ### This is an example Quick Start, with which you will deploy an existing
```
```suggestion
    ### In this example quick start, you'll deploy an existing image to OpenShift. For more information on writing quick starts, see our [quick start guidelines.](http://openshift.github.io/openshift-origin-design/conventions/documentation/quick-starts.html)
```

jhadvig commented:
@abigaeljamie I've addressed your comments :) PTAL

abigaeljamie commented on frontend/public/models/yaml-templates.ts:
```diff
@@ -1243,6 +1244,58 @@ metadata:
+    - title: Run the deployed application
+      description: |-
+        ### To run the **hello-openshift** deployment application, follow these steps:
+        1. In the **Topology** view, click the deployment's circle.
```
```suggestion
        1. In the **Topology** view, click the **hello-openshift** deployment circle. 
```

abigaeljamie commented on frontend/public/models/yaml-templates.ts:
```diff
@@ -1243,6 +1244,58 @@ metadata:
+      description: |-
+        ### To run the **hello-openshift** deployment application, follow these steps:
+        1. In the **Topology** view, click the deployment's circle.
+        2. In the **hello-openshift** deployment circle, click the **Open URL** icon, which represents the route. This opens the URL and runs the application.
```
```suggestion
        2. Click the **Open URL** icon, which represents the route. This opens the URL and runs the application.
```

abigaeljamie reviewed (approved):
Thanks, @jhadvig ! I tweaked steps 1 and 2 in the second task, otherwise it looks great. I don't need to review it again once those edits are made, so I went ahead and approved this. Cheers!

jhadvig commented:
/test e2e-gcp-console

bmignano reviewed (approved):
LGTM! Thanks, @jhadvig !

jhadvig commented:
/lgtm
/hold
/assign @ahardin-rh @yapei @sferich888

yapei commented:
@schituku Please give it a test, thanks

jhadvig commented:
/test e2e-gcp-console

ahardin-rh commented:
/label docs-approved

jhadvig commented:
Updated line#1270 with `[dropdown menu]{{highlight tour-perspective-dropdown}}`markdown based on @bmignano suggestion

yapei commented:
tested locally and no issues found from QE's point of view

yapei commented:
/label qe-approved

jhadvig commented:
/retest

jhadvig commented:
/retest

spadgett reviewed (approved):
/lgtm

jhadvig commented:
/retest

jhadvig commented:
/retest

jhadvig commented:
@sferich888 PTAL

reestr commented:
/label px-approved

spadgett commented:
/hold cancel
/override ci/prow/kubevirt-plugin
/retest
